### PR TITLE
[feature] [kubernetes] add loadBalancerSourceRanges to service-external.yaml for controller and broker

### PR DESCRIPTION
**Pull request by jameskelleher**
The primary purpose of this PR is to [add support for](https://github.com/apache/pinot/issues/9438) `loadBalancerSourceRanges` (which already exists in `values.yaml`) to the `service-external.yaml` templates for the controller and the broker. This allows the developer to specify which IP addresses they'd like to permit controller/broker access.

Additionally, I changed some template file suffixes from `.yml` to `.yaml`, to a) match internal consistency and b) adhere to Helm's recommendations.

# Testing
I tested the changes by running `helm install --dry-run --debug pinot ./kubernetes/helm/pinot`. For each test, I provide what I set `loadBalancerSourceRanges` to in `values.yaml`, as well as the templated result for `service-external.yaml`. 

## Test 1: Empty source range list
```yaml
# values.yaml
loadBalancerSourceRanges: []
```

```yaml
# Source: pinot/templates/controller/service-external.yaml
apiVersion: v1
kind: Service
metadata:
  name: pinot-controller-external
  annotations:
    {}
  labels:
    helm.sh/chart: pinot-0.2.6-SNAPSHOT
    app: pinot
    release: pinot
    app.kubernetes.io/version: "0.2.6-SNAPSHOT"
    app.kubernetes.io/managed-by: Helm
    heritage: Helm
    component: controller
spec:
  type: LoadBalancer
  ports:
    - name: external-controller
      port: 9000
  selector:
    app: pinot
    release: pinot
    component: controller
  loadBalancerSourceRanges:
        []
```

```yaml
# Source: pinot/templates/broker/service-external.yaml
apiVersion: v1
kind: Service
metadata:
  name: pinot-broker-external
  annotations:
        {}
  labels:
    helm.sh/chart: pinot-0.2.6-SNAPSHOT
    app: pinot
    release: pinot
    app.kubernetes.io/version: "0.2.6-SNAPSHOT"
    app.kubernetes.io/managed-by: Helm
    heritage: Helm
    component: broker
spec:
  type: LoadBalancer
  ports:
    - name: external-broker
      port: 8099
  selector:
    app: pinot
    release: pinot
    component: broker
  loadBalancerSourceRanges:
        []
```
## Test 2: Non-empty list
```yaml
# values.yaml
loadBalancerSourceRanges:
    - 127.0.0.1/32
    - 123.4.5.6/24
```

```yaml
# Source: pinot/templates/controller/service-external.yaml
apiVersion: v1
kind: Service
metadata:
  name: pinot-controller-external
  annotations:
    {}
  labels:
    helm.sh/chart: pinot-0.2.6-SNAPSHOT
    app: pinot
    release: pinot
    app.kubernetes.io/version: "0.2.6-SNAPSHOT"
    app.kubernetes.io/managed-by: Helm
    heritage: Helm
    component: controller
spec:
  type: LoadBalancer
  ports:
    - name: external-controller
      port: 9000
  selector:
    app: pinot
    release: pinot
    component: controller
  loadBalancerSourceRanges:
    - 127.0.0.1/32
    - 123.4.5.6/24
```

```yaml
# Source: pinot/templates/broker/service-external.yaml
apiVersion: v1
kind: Service
metadata:
  name: pinot-broker-external
  annotations:
    {}
  labels:
    helm.sh/chart: pinot-0.2.6-SNAPSHOT
    app: pinot
    release: pinot
    app.kubernetes.io/version: "0.2.6-SNAPSHOT"
    app.kubernetes.io/managed-by: Helm
    heritage: Helm
    component: broker
spec:
  type: LoadBalancer
  ports:
    - name: external-broker
      port: 8099
  selector:
    app: pinot
    release: pinot
    component: broker
  loadBalancerSourceRanges:
    - 127.0.0.1/32
    - 123.4.5.6/24
```

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
cc @xiangfu0 to take a look

**Review by walterddr — COMMENTED** [review]
thank you for your contribution. this looks great. i have one question to follow up otherwise lgtm

File: kubernetes/helm/pinot/templates/broker/service-external.yaml
Comment (jameskelleher) [reply kubernetes/helm/pinot/templates/broker/service-external.yaml]: @walterddr good catch, the dry run failed. I made an update that maintains the same behavior for the previous two tests, as well as handles the case you provided (`loadBalancerSourceRanges` will not be in the result template if not provided as a value)

**Review by walterddr — APPROVED** [review]
looks good thanks for the fix. cc @xiangfu0 to take another look before I merge it

File: kubernetes/helm/pinot/templates/broker/service-external.yaml
Comment (walterddr) [reply kubernetes/helm/pinot/templates/broker/service-external.yaml]: curious, is setting this config to empty list behave exactly the same as not setting this flag at all?

**Pull request merged**
